[GLIB] Modernize some argument coders

csaavedra opened the pull request:
#### 5fc7a2593e690290b50e9c9f8d30aab3374798d5
<pre>
[GLIB] Modernize some argument coders
<a href="https://bugs.webkit.org/show_bug.cgi?id=272983">https://bugs.webkit.org/show_bug.cgi?id=272983</a>

Reviewed by Michael Catanzaro.

Move some of the argument coders uses for GLib and GTK types
to use the modern ArgumentCoder API. Remove the GKeyFile coders,
as one can actually use GVariant for serialization of GTK printing
structs and we already have a GVariant serializer.

Also add UNLIKELY() for the std::nullopt bailout cases that can be
optimized.

* Source/WebKit/Platform/IPC/glib/ArgumentCodersGlib.h:
* Source/WebKit/Shared/glib/ArgumentCodersGLib.cpp:
(IPC::ArgumentCoder&lt;GRefPtr&lt;GVariant&gt;&gt;::decode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GTlsCertificate&gt;&gt;::decode):
(IPC::ArgumentCoder&lt;GTlsCertificateFlags&gt;::decode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GUnixFDList&gt;&gt;::decode):
* Source/WebKit/Shared/gtk/ArgumentCodersGtk.cpp:
(IPC::ArgumentCoder&lt;GRefPtr&lt;GtkPrintSettings&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GtkPrintSettings&gt;&gt;::decode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GtkPageSetup&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GtkPageSetup&gt;&gt;::decode):
(IPC::encodeGKeyFile): Deleted.
(IPC::decodeGKeyFile): Deleted.

Canonical link: <a href="https://commits.webkit.org/277967@main">https://commits.webkit.org/277967@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b2e9fd6d64bf6a9204f3b9cccf6e1e0ffd29a2f5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48998 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28211 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51965 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51686 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45068 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34179 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25739 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40053 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/50027 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25855 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42242 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21161 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23313 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7053 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45239 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43916 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53596 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24049 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20302 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47365 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25312 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42450 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46331 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10799 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26121 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25032 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
Are you sure that passing empty strings is truly unlikely? I ask because (a) these can have a deleterious performance impact if the condition does evaluate to true, and (b) I don't see other ArgumentCoders files using UNLIKELY like this.

csaavedra committed:
[GLIB] Modernize some argument coders
https://bugs.webkit.org/show_bug.cgi?id=272983

Reviewed by Michael Catanzaro.

Move some of the argument coders uses for GLib and GTK types
to use the modern ArgumentCoder API. Remove the GKeyFile coders,
as one can actually use GVariant for serialization of GTK printing
structs and we already have a GVariant serializer.

Also add UNLIKELY() for the std::nullopt bailout cases that can be
optimized.

* Source/WebKit/Platform/IPC/glib/ArgumentCodersGlib.h:
* Source/WebKit/Shared/glib/ArgumentCodersGLib.cpp:
(IPC::ArgumentCoder<GRefPtr<GVariant>>::decode):
(IPC::ArgumentCoder<GRefPtr<GTlsCertificate>>::decode):
(IPC::ArgumentCoder<GTlsCertificateFlags>::decode):
(IPC::ArgumentCoder<GRefPtr<GUnixFDList>>::decode):
* Source/WebKit/Shared/gtk/ArgumentCodersGtk.cpp:
(IPC::ArgumentCoder<GRefPtr<GtkPrintSettings>>::encode):
(IPC::ArgumentCoder<GRefPtr<GtkPrintSettings>>::decode):
(IPC::ArgumentCoder<GRefPtr<GtkPageSetup>>::encode):
(IPC::ArgumentCoder<GRefPtr<GtkPageSetup>>::decode):
(IPC::encodeGKeyFile): Deleted.
(IPC::decodeGKeyFile): Deleted.

Canonical link: https://commits.webkit.org/277967@main
